README.md
# Adf.ly Laravel API wrapper

[![Latest Version on Packagist](https://img.shields.io/packagist/v/toneflix-code/adf-ly-laravel.svg?style=flat-square)](https://packagist.org/packages/toneflix-code/adf-ly-laravel)
[![Total Downloads](https://img.shields.io/packagist/dt/toneflix-code/adf-ly-laravel.svg?style=flat-square)](https://packagist.org/packages/toneflix-code/adf-ly-laravel)

[![Laravel8|9](https://img.shields.io/badge/Laravel-8|9-orange.svg)](http://laravel.com)
[![License](https://img.shields.io/github/license/mashape/apistatus.svg)](https://packagist.org/packages/toneflix-code/adf-ly-laravel)

This library is a simple laravel package that wraps around Adf.ly API implementations.

Please refere to [Adf.ly API Documentation](https://login.adf.ly/static/other/adfly_api_v1_documentation.pdf?v=20191108) for detailed api use description.

## Requirements

- [PHP >= 7.41](http://php.net/)
- [Laravel 8|9](https://github.com/laravel/framework)

## Installation

You can install the package via composer:

```bash
composer require toneflix-code/adf-ly-laravel
```

#### Service Provider & Facade (Optional on Laravel 5.5+)

Register provider and facade on your `config/app.php` file.

```php
'providers' => [
    ...,
    ToneflixCode\AdfLy\AdfLyServiceProvider::class,
]

'aliases' => [
    ...,
    'AdfLy' => ToneflixCode\AdfLy\AdfLyFacade::class,
]
```

#### Configuration (Optional)

```bash
php artisan vendor:publish --provider="ToneflixCode\AdfLy\AdfLyServiceProvider"
```

## API Keys

To start using this library you are required to configure your API keys in your .env file with these variables

```
ADFLY_SECRET_KEY=your-adf.ly secret key
ADFLY_PUBLIC_KEY=your-adf.ly public key
ADFLY_USER_ID=your-adf.ly user ID
```

## Usage

### Shorten Url

```php
use ToneflixCode\AdfLy\AdfLy;
$adfly = new AdfLy;
$res = $adfly->shorten(array('http://stackoverflow.com/users'), 'q.gs');

$shortenedUrl1 = $res['data'][0];
$hash1 = substr($shortenedUrl1['short_url'],strrpos($shortenedUrl1['short_url'],'/')+1);
echo 'First URL shortened (' . $hash1 . '): ' . print_r($res);

$res = $adfly->shorten(array('http://www.reddit.com'), 'q.gs');
$shortenedUrl2 = $res['data'][0];
$hash2 = substr($shortenedUrl2['short_url'],strrpos($shortenedUrl2['short_url'],'/')+1);
echo 'Another URL shortened (' . $hash2 . '): ' . print_r($res);

$res = $adfly->shorten(array('www.youtube.com'), 'q.gs', 'banner');
$shortenedUrl3 = $res['data'][0];
echo 'Another URL shortened: ' . print_r($res);

$res = $adfly->shorten(array('http://www.len10.com/videos/'), 'q.gs', 'int', 13);
$shortenedUrl4 = $res['data'][0];
echo 'Another URL shortened: ' . print_r($res);
```

#### Parameters

```php
function shorten(
    array $urls,
    $domain = false,
    $advertType = false,
    $groupId = false,
    $title = false,
    $customName = false
)
```

## Expand.

```php
print_r($adfly->expand(array($shortenedUrl3['short_url'],$shortenedUrl4['short_url']),array($hash1,$hash2)));
```

## LISTING

```php
// List Urls
$urlList = $adfly->getUrls();
print_r($urlList);

// Update Url
$adfly->updateUrl($shortenedUrl1['id'], 'http://modifiedurlaaaa.cat', "int", "The  updated URL", 13, false, false);
print_r($adfly->expand(array(),array($hash1)));

foreach($urlList['data'] as $url){
    $adfly->deleteUrl($url['id']);
}

//List Urls again
$urlList = $adfly->getUrls();
print_r($urlList);
```

## GROUPS

```php
$g = $adfly->createGroup('API Group');
print_r($g);

$g = $adfly->getGroups(1);
print_r($g);
```

## REFERRERS GET

```php
$res = $adfly->getReferrers();
print_r($res);
```

## COUNTRIES GET

```php
$res = $adfly->getCountries();
print_r($res);
```

## ANNOUNCEMENTS GET

```php
$res = $adfly->getAnnouncements();
print_r($res);
```

## publisherReferralStats GET

```php
$res = $adfly->getPublisherReferrals();
print_r($res);
```

## advertiserReferralStats GET

```php
$res = $adfly->getAdvertiserReferrals();
print_r($res,1);
```

## withdrawalTransactions GET

```php
$res = $adfly->getWithdrawalTransactions();
print_r($res,1);
```

## withdraw GET

```php
$res = $adfly->getWithdraw();
print_r($res,1);
```

## withdraw request GET

```php
$res = $adfly->withdrawRequestInitiate();
print_r($res,1);
```

## withdraw request DELETE

```php
$res = $adfly->withdrawRequestCancel();
print_r($res,1);
```

## publisher Stats GET

```php
$res = $adfly->getPublisherStats();
print_r($res,1);
```

## user Profile GET

```php
$res = $adfly->getProfile();
print_r($res,1);
```

## advertiser Campaigns GET

```php
$res = $adfly->getAdvertiserCampaigns();
print_r($res,1);
```

## advertiser Graph GET

```php
$res = $adfly->getAdvertiserGraph(null,156);
print_r($res,1);
```

## advertisers Campaign parts GET

```php
$res = $adfly->getAdvertiserCampaignParts(739026);
print_r($res,1);
```

## auth POST

```php
$res = $adfly->auth('1', '2');
print_r($res,1);
```

## account publisher referrals

```php
$res = $adfly->getAccountPubReferrals('', '', 1);
print_r($res);
```

## account advertiser referrals

```php
$res = $adfly->getAccountAdvReferrals('', '', 1);
print_r($res);
```

## account popad referrals

```php
$res = $adfly->getAccountPopReferrals('', '', 1);
print_r($res);
```

## account total referrals

```php
$res = $adfly->getAccountTotalReferrals();
print_r($res);
```

## Domains

```php
$res = $adfly->getDomains();
print_r($res);
```

## update account details

```php
$res = $adfly->updateAccountDetails([]);
print_r($res);
```

## update account password

```php
$res = $adfly->updatePassword('oldpassword', 'newpassword', 'newpassword');
print_r($res);
```

## get account countries

```php
$res = $adfly->getAccountCountries();
print_r($res);
```

## get account details

```php
$res = $adfly->getAccountDetails();
print_r($res);
```

### Testing

```bash
composer test
```

### Changelog

Please see [CHANGELOG](CHANGELOG.md) for more information what has changed recently.

## Contributing

Please see [CONTRIBUTING](CONTRIBUTING.md) for details.

### Security

If you discover any security related issues, please email code@toneflix.com.ng instead of using the issue tracker.

## Credits

- [Toneflix Code](https://github.com/toneflix)
- [All Contributors](../../contributors)

## License

The MIT License (MIT). Please see [License File](LICENSE.md) for more information.


CHANGELOG.md
# Changelog

All notable changes to `adf-ly-laravel` will be documented in this file

## 1.0.0 - 201X-XX-XX

- initial release


src/example.php
<?php
	require_once '../../Examples/PHP/PhpExample.php';
		/**
		 * Example script entry point.
		 */
		function main() {
            define('LB',"<br />\r\n");

            $ex = new PhpExample(2, '2ba3f6ce601d043c177eb2a83eb34f5f', '4c8fa05a-d826-4c06-86e4-59b86bf4868c');

//             //echo "Starting the demo...".LB.LB;
//             echo "SHORTENING...".LB;

// 			// Shorten Url
//             $res = $ex->shorten(array('http://stackoverflow.com/users'), 'adfly.local');
//             $shortenedUrl1 = $res['data'][0];
//             $hash1 = substr($shortenedUrl1['short_url'],strrpos($shortenedUrl1['short_url'],'/')+1);
//             echo 'First URL shortened (' . $hash1 . '): ' . print_r($res,1) . LB;

//             $res = $ex->shorten(array('http://www.reddit.com'), 'q.gs');
//             $shortenedUrl2 = $res['data'][0];
//             $hash2 = substr($shortenedUrl2['short_url'],strrpos($shortenedUrl2['short_url'],'/')+1);
//             echo 'Another URL shortened (' . $hash2 . '): ' . print_r($res,1) . LB;

//             $res = $ex->shorten(array('www.youtube.com'), 'q.gs', 'banner');
//             $shortenedUrl3 = $res['data'][0];
//             echo 'Another URL shortened: ' . print_r($res,1) . LB;

//             $res = $ex->shorten(array('http://www.len10.com/videos/'), 'q.gs', 'int', 13);
//             $shortenedUrl4 = $res['data'][0];
//             echo 'Another URL shortened: ' . print_r($res,1) . LB;

//             echo LB."EXPAND...".LB;
//             // Expand examples.
//             echo 'Shortened URLS just created: ' . print_r($ex->expand(array($shortenedUrl3['short_url'],$shortenedUrl4['short_url']),array($hash1,$hash2)),1);

//             echo LB."LISTING...".LB;
//             //List Urls
//             $urlList = $ex->getUrls();
//             echo 'Listing page 1 URLS...: ' . print_r($urlList,1);

//             //Update Url
//             echo LB."UPDATING LINK...".LB;
//             $ex->updateUrl($shortenedUrl1['id'], 'http://modifiedurlaaaa.cat', "int", "The  updated URL", 13, false, false);
//             echo 'Modified URL: ' . print_r($ex->expand(array(),array($hash1)),1);

//             foreach($urlList['data'] as $url){
//                 echo 'Deleting URL ID: ' . $url['id'] . LB;
//                 $ex->deleteUrl($url['id']);
//             }

//             echo LB."LISTING AGAIN...".LB;
//             //List Urls
//             $urlList = $ex->getUrls();
//             echo 'Listing page 1 URLS...: ' . print_r($urlList,1) . LB;

//             //GROUPS
//             echo LB."GROUPS".LB;
//             $g = $ex->createGroup('API Group');
//             echo 'Created group: ' . print_r($g,1).LB;

//             $g = $ex->getGroups(1);
//             echo 'Listing page 1 GROUPS...: ' . print_r($g,1).LB;

//             //REFERRERS GET
//             echo LB."REFERRERS".LB;
//             $res = $ex->getReferrers();
//             echo 'URL Referrers: ' . print_r($res,1).LB;

//             //COUNTRIES GET
//             echo LB."COUNTRIES".LB;
//             $res = $ex->getCountries();
//             echo 'URL Countries: ' . print_r($res,1).LB;

//             //ANNOUNCEMENTS GET
//             echo LB."ANNOUNCEMENTS".LB;
//             $res = $ex->getAnnouncements();
//             echo 'Announcements: ' . print_r($res,1).LB;

//             //publisherReferralStats GET
//             echo LB."PUBLISHER REFERRAL STATS".LB;
//             $res = $ex->getPublisherReferrals();
//             echo 'Stats: ' . print_r($res,1).LB;

//             //advertiserReferralStats GET
//             echo LB."ADVERTISER REFERRAL STATS".LB;
//             $res = $ex->getAdvertiserReferrals();
//             echo 'Stats: ' . print_r($res,1).LB;

//             ///withdrawalTransactions GET
//             echo LB."WITHDRAWAL TRANSACTRIONS".LB;
//             $res = $ex->getWithdrawalTransactions();
//             echo 'Transactions: ' . print_r($res,1).LB;

//             ///withdraw GET
//             echo LB."WITHDRAW".LB;
//             $res = $ex->getWithdraw();
//             echo 'Data: ' . print_r($res,1).LB;

//             ///withdraw request GET
//             echo LB."WITHDRAW REQUEST START".LB;
//             $res = $ex->withdrawRequestInitiate();
//             echo 'Data: ' . print_r($res,1).LB;

//             ///withdraw request DELETE
//             echo LB."WITHDRAW REQUEST CANCEL".LB;
//             $res = $ex->withdrawRequestCancel();
//             echo 'Data: ' . print_r($res,1).LB;


//             //publisherStats GET
//             echo LB."PUBLISHER STATS".LB;
//             $res = $ex->getPublisherStats();
//             echo 'Stats: ' . print_r($res,1).LB;

//             //user Profile GET
//             echo LB."USER PROFILE".LB;
//             $res = $ex->getProfile();
//             echo 'User: ' . print_r($res,1).LB;

//             //advertiserStats GET
//             echo LB."ADVERTISER STATS".LB;
//             $res = $ex->getAdvertiserCampaigns();
//             echo 'Stats: ' . print_r($res,1).LB;

//             //advertiserStats GET
//             echo LB."ADVERTISER GRAPH".LB;
//             $res = $ex->getAdvertiserGraph(null,156);
//             echo 'Stats: ' . print_r($res,1).LB;

//             //advertiserStats GET
//             echo LB."ADVERTISER STATS".LB;
//             $res = $ex->getAdvertiserCampaignParts(739026);
//             echo 'Stats: ' . print_r($res,1).LB;

//             //auth POST
//             echo LB."AUTH".LB;
//             $res = $ex->auth('1', '2');
//             echo 'Auth: ' . print_r($res,1).LB;

            // account publisher referrals
//             echo LB."ACCOUNT PUBLISHER REFERRALS".LB;
//             $res = $ex->getAccountPubReferrals('', '', 1, 1);
// 			echo 'Referrals: ' . print_r($res, 1).LB;

            // account advertiser referrals
//              echo LB."ACCOUNT ADVERTISER REFERRALS".LB;
//              $res = $ex->getAccountAdvReferrals('', '', 1, 1);
//  			echo 'Referrals: ' . print_r($res, 1).LB;

            // account popad referrals
//             echo LB."ACCOUNT POPAD REFERRALS".LB;
//             $res = $ex->getAccountPopReferrals('', '', 1, 1);
// 			echo 'Referrals: ' . print_r($res, 1).LB;

            // account total referrals
//              echo LB."ACCOUNT TOTAL REFERRALS".LB;
//              $res = $ex->getAccountTotalReferrals();
//  			echo 'Referrals: ' . print_r($res, 1).LB;

            // account popad referrals
//              echo LB."DOMAINS".LB;
//              $res = $ex->getDomains();
//  			echo 'Domains: ' . print_r($res, 1).LB;

            // update account details
//             echo LB."UDATE ACCOUNT DETAILS".LB;
//             $res = $ex->updateAccountDetails([]);
//             echo 'Result: ' . print_r($res, 1).LB;

            // update account details
//             echo LB."UDATE ACCOUNT PASSWORD".LB;
//             $res = $ex->updatePassword('oldpassword', 'newpassword', 'newpassword');
//             echo 'Result: ' . print_r($res, 1).LB;

           // get account countries
//              echo LB."ACCOUNT COUNTRIES".LB;
//              $res = $ex->getAccountCountries();
//              echo 'Result: ' . print_r($res, 1).LB;

            // get account countries
//              echo LB."ACCOUNT DETAILS".LB;
//              $res = $ex->getAccountDetails();
//              echo 'Result: ' . print_r($res, 1).LB;
        }

	// Runs usage examples.
	main();
